src/app/components/AppNavigator/index.tsx

```diff
@@ -0,0 +1,96 @@
+  stackNavigatorConfig,
+  tabBarOptions,
+  wishlistStackConfig
+} from '@app/config/navigation';
+import HomeScreen from '@screens/HomeScreen';
+import DetailScreen from '@screens/DetailScreen';
+import ComingSoonScreen from '@screens/ComingSoonScreen';
+
+const LibraryStackNavigator = createStackNavigator();
+const WishlistStackNavigator = createStackNavigator();
+const AddNewStackNavigator = createStackNavigator();
+const RentalsStackNavigator = createStackNavigator();
+const SettingsStackNavigator = createStackNavigator();
+const TabNavigator = createBottomTabNavigator();
+function LibraryStackScreen() {
+  return (
+    <LibraryStackNavigator.Navigator initialRouteName={HOME_SCREEN} screenOptions={stackNavigatorConfig}>
+      <LibraryStackNavigator.Screen name={HOME_SCREEN} component={HomeScreen} />
+      <LibraryStackNavigator.Screen name={DETAIL_SCREEN} component={DetailScreen} />
+    </LibraryStackNavigator.Navigator>
+  );
+}
+function WishlistStackScreen() {
+  return (
+    <WishlistStackNavigator.Navigator
+      initialRouteName={COMING_SOON_SCREEN}
+      screenOptions={stackNavigatorConfig}>
+      <WishlistStackNavigator.Screen name={COMING_SOON_SCREEN} component={ComingSoonScreen} />
+    </WishlistStackNavigator.Navigator>
+  );
+}
+function AddNewStackScreen() {
+  return (
+    <AddNewStackNavigator.Navigator
+      initialRouteName={COMING_SOON_SCREEN}
+      screenOptions={stackNavigatorConfig}>
+      <AddNewStackNavigator.Screen name={COMING_SOON_SCREEN} component={ComingSoonScreen} />
+    </AddNewStackNavigator.Navigator>
+  );
+}
+function RentalStackScreen() {
+  return (
+    <RentalsStackNavigator.Navigator
+      initialRouteName={COMING_SOON_SCREEN}
+      screenOptions={stackNavigatorConfig}>
+      <RentalsStackNavigator.Screen name={COMING_SOON_SCREEN} component={ComingSoonScreen} />
+    </RentalsStackNavigator.Navigator>
+  );
+}
+function SettingsStackScreen() {
+  return (
+    <SettingsStackNavigator.Navigator
+      initialRouteName={COMING_SOON_SCREEN}
+      screenOptions={stackNavigatorConfig}>
+      <SettingsStackNavigator.Screen name={COMING_SOON_SCREEN} component={ComingSoonScreen} />
+    </SettingsStackNavigator.Navigator>
+  );
+}
+function AppNavigator() {
+  return (
+    <TabNavigator.Navigator tabBarOptions={tabBarOptions}>
+      <TabNavigator.Screen options={libraryStackConfig} name={LIBRARY_STACK} component={LibraryStackScreen} />
+      <TabNavigator.Screen
+        options={wishlistStackConfig}
+        name={WISHLIST_STACK}
+        component={WishlistStackScreen}
+      />
+      <TabNavigator.Screen options={addNewStackConfig} name={ADD_NEW_STACK} component={AddNewStackScreen} />
+      <TabNavigator.Screen options={rentalsStackConfig} name={RENTAL_STACK} component={RentalStackScreen} />
+      <TabNavigator.Screen
+        options={settingsStackConfig}
+        name={SETTINGS_STACK}
+        component={SettingsStackScreen}
+      />
+    </TabNavigator.Navigator>
```

Comment:
It's not necessary to create a stack for each tab, you could use the screen directly:
```
    <TabNavigator.Navigator
      initialRouteName={LIBRARY_STACK}
      screenOptions={{ tabBarIcon: TabBarIcon }}
      tabBarOptions={tabBarOptions}>
      <TabNavigator.Screen name={LIBRARY_STACK} component={LibraryStackScreen} options={libraryStackConfig} />
      <TabNavigator.Screen name={WISHLIST_STACK} component={ComingSoonScreen} options={wishlistStackConfig} />
      <TabNavigator.Screen name={ADD_NEW_STACK} component={ComingSoonScreen} options={addNewStackConfig} />
      <TabNavigator.Screen name={RENTAL_STACK} component={ComingSoonScreen} options={rentalsStackConfig} />
      <TabNavigator.Screen name={SETTINGS_STACK} component={ComingSoonScreen} options={settingsStackConfig} />
    </TabNavigator.Navigator>
```

Reply:
yeah, but I assumed each tab will need a stack in the future